core/lib/eth_client/src/types.rs

```diff
@@ -160,6 +163,87 @@ pub struct FailureInfo {
     pub gas_limit: U256,
 }
 
+/// The block type returned from RPC calls.
+/// This is generic over a `TX` type.
+/// We can't use `Block` from `web3` crate because it doesn't support `excessBlobGas`.
+#[derive(Debug, Default, Clone, PartialEq, Deserialize, Serialize)]
+pub struct Block<TX> {
```

Comment:
let's add at least one test to check that parsing from JSON works as expected (that we didn't do any typo in serde names for example)